src/irmin-pack/inode.ml

```diff
@@ -904,10 +917,21 @@ struct
                 | None -> true
                 | Some t ->
                     let t = target_of_ptr t in
-                    (if stable then not t.stable else true) && check t stable)
+                    (* FIXME: stable is true if t.kind = Node or an ancestor is stable
+                       but if stable is true we check if t.kind is Inode
+                       From
+                       if stable then match t.kind with Inode -> true | Node -> false else true
+                       To
+                       if stable then not (is_node t) else true
+                       To
+                       not stable || not (is_node t)
+                       To
+                       not (stable && is_node t)
+                    *)
+                    (not (stable && is_node t)) && check t stable)
```

Comment:
This is more compact and, I think, also clearer: we check that it is _false_ that there is a stable ancestor and that the current `t` is a stable.
I would remove the doc and replace with a short explanation, and maybe rename the `stable` variable into `any_stable_ancestor`?

Reply:
Yes, I tried to make what we were checking clearer and I think you got it perfectly right :-) I agree that changing the variable name would make it even more clear